NDTensors/src/dense/dense.jl

```diff
@@ -18,8 +18,24 @@ struct Dense{ElT,VecT<:AbstractArray} <: TensorStorage{ElT}
   function Dense{ElT,VecT}(data::Array) where {ElT,VecT<:AbstractArray{ElT}}
     return new{ElT,VecT}(vec(data))
   end
+
+  function Dense{ElT, VecT}() where {ElT, VecT<:AbstractArray{ElT}}
+    return new{ElT, VecT}(VecT(undef, 0))
+  end
 end
 
+## We need a general typeinfo constructor for Dense
+## This constructor will be called by all other versions
+Dense(eltype::Type, datatype::Type) = default_storage_type(Dense, eltype, datatype)
+
+Dense(datatype::Type{<:AbstractArray}) = Dense(Float64, datatype)
```

Comment:
This looks like it would be wrong for inputs like `Dense(Vector{Float32})`.

Maybe this should be:
```julia
dense_type(datatype::Type{<:AbstractArray}) = adapt_type(datatype, Dense)
```
or:
```julia
dense_type(datatype::Type{<:AbstractArray}) = set_datatype(Dense, datatype)
```